src/orion/benchmark/benchmark_client.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.benchmark.client` -- Benchmark client
+================================================================
+
+.. module:: client
+   :platform: Unix
+   :synopsis: Client to use Orion benchmark.
+
+"""
+import datetime
+
+from orion.benchmark import Benchmark, Study
+from orion.benchmark.assessment.base import BenchmarkAssessment
+from orion.benchmark.task.base import BenchmarkTask
+from orion.core.utils.exceptions import NoConfigurationError
+from orion.storage.base import get_storage, setup_storage
+
+
+def get_or_create_benchmark(
+    name, algorithms=None, targets=None, storage=None, debug=False
+):
+    """
+    Create or get a benchmark object.
+
+    Parameters
+    ----------
+    name: str
+        Name of the benchmark
+    algorithms: list, optional
+        Algorithms used for benchmark, each algorithm can be a string or dict.
+    targets: list, optional
+        Targets for the benchmark, each target will be a dict with two keys.
+
+        assess: list
+            Assessment objects
+        task: list
+            Task objects
+    storage: dict, optional
+        Configuration of the storage backend.
+    debug: bool, optional
+        If using in debug mode, the storage config is overrided with legacy:EphemeralDB.
+        Defaults to False.
+
+    Returns
+    -------
+    An instance of `orion.benchmark.Benchmark`
+    """
+    setup_storage(storage=storage, debug=debug)
+
+    # fetch benchmark from db
+    db_config = _fetch_benchmark(name)
+
+    benchmark_id = None
+
+    if db_config:
+        benchmark_id, algorithms, targets = _resolve_db_config(db_config)
+
+    if not algorithms or not targets:
+        raise NoConfigurationError(
+            "Benchmark {} does not exist in DB, "
+            "algorithms and targets space was not defined.".format(name)
+        )
+
+    benchmark = _create_benchmark(name, algorithms, targets)
+
+    if benchmark_id is None:
```

Comment:
Just to be sure I didn't miss anything. If we pass a benchmark config with a name, and the benchmark is already registered in DB but the passed config is different, the one from the config would have precedence and would be the one loaded and executed. Is that right? I think it would be important to check any differences and at least raise a warning in such case.

Reply:
Yes, there is no branching support like experiment and the existing db config will be honored not what different algorithms or targets for the second call.  I think we can raise a generic warning and print the db config when we detect there is difference.

Reply:
Yeah we should not aim to support an equivalent of the EVC for the benchmarks. I agree with a generic warning and db config print.